lib/spack/spack/build_systems/autotools.py

```diff
@@ -191,6 +191,111 @@ def archive_files(self) -> List[str]:
             files.append(self._removed_la_files_log)
         return files
 
+    @property
```

Comment:
All other files LGTM. Here, can we avoid shuffling code around this much?

Reply:
It's just grouping `@run_before` / `@run_after` and sorting them in the order they run, plus making them _private. I can split that out if you want

Reply:
It's just the sorting that gives the huge diff. Making them private is :100:

Reply:
Okay, I'll split it up.